sqlmesh/core/context_diff.py

```diff
@@ -172,7 +177,9 @@ def create(
             is_unfinalized_environment=bool(env and not env.finalized_ts),
             create_from=create_from,
             added=added,
-            removed=removed,
+            removed_snapshots={
+                name: info for name, info in existing_info.items() if name in removed
```

Comment:
[Nit] Considering that we expect `removed` to be empty most of the time I think you're better off iterating over `removed` instead and then getting the `info` instance from the `existing_info` dict.